Add patch to use cmake's FindThreads and use win32 threads on Windows

ryanvolz opened the pull request:
The pthreads-win32 library conflicts with the new mingw stack used for other packages, so this improves compatibility with those new packages.

<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* ~[ ] Reset the build number to `0` (if the version changed)~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

ryanvolz committed:
Bump build number

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ This recipe is using a compiler, which now requires adding a build dependence on `{{ stdlib("c") }}` as well. Note that this rule applies to each output of the recipe using a compiler. For further details, please see https://github.com/conda-forge/conda-forge.github.io/issues/2102.

For **recipe/meta.yaml**:

 * ℹ️ The recipe is not parsable by parser `conda-recipe-manager`. The recipe can only be automatically migrated to the new v1 format if it is parseable by conda-recipe-manager.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/13771594334](https://github.com/conda-forge/conda-forge-webservices/actions/runs/13771594334). Examine the logs at this URL for more detail.</sub>

ryanvolz committed:
Add stdlib('c') dependency

ryanvolz committed:
Remove jinja for loop for parseability by conda-recipe-manager

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

ryanvolz committed:
Replace pthreads-win32 with winpthreads-devel for pthreads

pthreads-win32 is not compatible with the pthreads provided by mingw
(winpthreads-devel), and since some packages need to be built with mingw
and used its pthreads and we want to be able to install this package
alongside those, switch to winpthreads-devel.

ryanvolz committed:
MNT: Re-rendered with conda-build 25.1.1, conda-smithy 3.47.0, and conda-forge-pinning 2025.03.10.16.49.26

ryanvolz committed:
libmirisdr4 package does not use winpthreads, remove dep

conda-forge-admin commented:
Hi! This is the friendly conda-forge automerge bot!

I considered the following status checks when analyzing this PR:
 - **linter**: passed
 - **azure**: passed


Thus the PR was passing and merged! Have a great day!
